awslambda/__init__.py

```diff
@@ -0,0 +1,183 @@
+        os.path.join(current_dir_path, "config.yml"),
+        os.path.join(project_path, "config.yml"),
+    )
+
+    # Copy app.py which handles the Lambda events
+    shutil.copy(
+        os.path.join(current_dir_path, "app.py"), os.path.join(project_path, "app.py")
+    )
+
+
+def generate_aws_compatible_string(*items, max_length=63):
+    """
+    Generate a AWS resource name that is composed from list of string items. This
+    function replaces all invalid characters in the given items into '-', and allow user
+    to specify the max_length for each part separately by passing the item and its max
+    length in a tuple, e.g.:
+
+    >> generate_aws_compatible_string("abc", "def")
+    >> 'abc-def'  # concatenate multiple parts
+
+    >> generate_aws_compatible_string("abc_def")
+    >> 'abc-def'  # replace invalid chars to '-'
+
+    >> generate_aws_compatible_string(("ab", 1), ("bcd", 2), max_length=4)
+    >> 'a-bc'  # trim based on max_length of each part
+    """
+    trimmed_items = [
+        item[0][: item[1]] if type(item) == tuple else item for item in items
+    ]
+    items = [item[0] if type(item) == tuple else item for item in items]
+
+    for i in range(len(trimmed_items)):
+        if len("-".join(items)) <= max_length:
+            break
+        else:
+            items[i] = trimmed_items[i]
+
+    name = "-".join(items)
+    if len(name) > max_length:
+        raise Exception(
+            "AWS resource name {} exceeds maximum length of {}".format(name, max_length)
+        )
+    invalid_chars = re.compile("[^a-zA-Z0-9-]|_")
+    name = re.sub(invalid_chars, "-", name)
+    return name
+
+
+def generate_lambda_resource_names(name):
+    sam_template_name = generate_aws_compatible_string(f"{name}-template")
+    deployment_stack_name = generate_aws_compatible_string(f"{name}-stack")
+    repo_name = generate_aws_compatible_string(f"{name}-repo")
+
+    return sam_template_name, deployment_stack_name, repo_name
+
+
+def generate_docker_image_tag(registry_uri, bento_name, bento_version):
+    image_tag = f"{bento_name}-{bento_version}".lower()
+    return f"{registry_uri}:{image_tag}"
+
+
+def generate_aws_lambda_cloudformation_template_file(
+    project_dir,
+    api_names,
+    bento_service_name,
+    docker_tag,
+    docker_file,
+    docker_context,
+    memory_size: int,
+    timeout: int,
+):
+    template_file_path = os.path.join(project_dir, "template.yaml")
+    yaml = YAML()
+    sam_config = {
+        "AWSTemplateFormatVersion": "2010-09-09",
+        "Transform": "AWS::Serverless-2016-10-31",
+        "Globals": {
+            "Function": {"Timeout": timeout, "MemorySize": memory_size},
+            # "Api": {
+            # "BinaryMediaTypes": ["image~1*"],
+            # "Cors": "'*'",
+            # "Auth": {
+            # "ApiKeyRequired": False,
+            # "DefaultAuthorizer": "NONE",
+            # "AddDefaultAuthorizerToCorsPreflight": False,
+            # },
+            # },
```

Comment:
Any reason why we comment out these fields?

Reply:
Yeah, I wanted to ask you what this section does. I took this from the existing lambda deployment but the API works when I tested it without this. Should I look up exactly what these options do and add them up? I didn't understand the use case this had in the previous lambda deployment part.

Reply:
Most of this is for the API gateway. For BentoService are predicting images, we will need to make sure the binary type is allowed by the API gateway

Reply:
ohh got it! wasn't testing on Images. I'll look up about these args and add them